site/app/libraries/Utils.php

```diff
@@ -240,32 +240,42 @@ public static function getCompareByReference() {
      * students_version is an array of user and their highest submitted version
      */
 
-    public static function getAutoFillData($students, $students_version = null){
+    public static function getAutoFillData($students, $students_version = null) {
         $students_full = array();
         $null_section = array();
         foreach ($students as $student) {
             $student_entry = array('value' => $student->getId(),
                     'label' => $student->getDisplayedFirstName() . ' ' . $student->getDisplayedLastName() . ' <' . $student->getId() . '>');
-            $students_full[] = $student_entry;
-            if($students_version != null){
-                if($student->getRegistrationSection() != null && array_key_exists($student->getId(),$students_version)){
+            if($students_version != null) {
+                if($student->getRegistrationSection() != null && array_key_exists($student->getId(),$students_version)) {
                     if ($students_version[$student->getId()] !== 0) {
                         $student_entry['label'] .= ' (' .
                         $students_version[$student->getId()] . ' Prev Submission)';
                     }
                 }
-            }else{
+                $students_full[] = $student_entry;
```

Comment:
Moving `$students_full[] = $student_entry;` to here does not solve the problem. It just makes `$students_full` empty. You could test by entering the id of a student with a registration section, and you will find he/she will be categorized as `[NULL section]` under your code.

The root of this problem is that, if `$students_version` is null, then the code jumps to line 258, marking everything as `[NULL section]` without asking if `$student->getRegistrationSection() != null`.

An ugly solution can be adding an `elseif` in line 257:

```
elseif ($student->getRegistrationSection() != null) {
        $students_full[] = $student_entry;
}
```

You could surely come up with a more elegant solution, probably by refactoring the `if-else` logic here.